modules/ui/rapid_view_manage_datasets.js

```diff
@@ -154,17 +123,70 @@ export function uiRapidViewManageDatasets(context, parentModal) {
       .attr('target', '_blank');
 
 
-    /* Pages section */
-    let pagesSection = selection.selectAll('.rapid-view-manage-pages')
-      .data([0]);
+    /* Filter section */
+    let filterEnter = selection.selectAll('.rapid-view-manage-filter')
+      .data([0])
+      .enter()
+      .append('div')
+      .attr('class', 'modal-section rapid-view-manage-filter');
 
-    let pagesSectionEnter = pagesSection.enter()
+
+    let filterSearchEnter = filterEnter
       .append('div')
-      .attr('class', 'modal-section rapid-view-manage-pages');
+      .attr('class', 'rapid-view-manage-filter-search-wrap');
+
+    filterSearchEnter
+      .call(svgIcon('#fas-filter', 'inline'));
 
-    pagesSection = pagesSection
-      .merge(pagesSectionEnter)
-      .call(renderPages);
+    filterSearchEnter
+      .append('input')
+      .attr('class', 'rapid-view-manage-filter-search')
+      .attr('placeholder', t('rapid_feature_toggle.esri.filter_datasets'))
+      .call(utilNoAuto)
+      .on('input', d3_event => {
+        const target = d3_event.target;
+        const val = (target && target.value) || '';
+        _filterText = val.trim().toLowerCase();
+        dsSection.call(renderDatasets);
+      });
+
+
+    let filterTypeEnter = filterEnter
+      .append('div')
+      .attr('class', 'rapid-view-manage-filter-type-wrap');
+
+    filterTypeEnter
+      .append('input')
+      .attr('class', 'rapid-view-manage-filter-type')
```

Comment:
Building on this- if the user enters some text in the 'type' field, hits enter, sees '0 dataset(s) found', and then types some more into the text field, we should give them a visual cue that they haven't yet hit 'entier' yet. perhaps we grey out the '0 datasets found' and / or replace it with an entreaty to 'hit enter'? dunno. Moot point if we just want to eliminate the free-form text edit nature of the control.